mpas_analysis/shared/timekeeping/utility.py

```diff
@@ -420,7 +420,7 @@ def _parse_date_string(dateString, isInterval=False):  # {{{
         offset = 1
 
     # change underscores to spaces so both can be supported
-    dateString = dateString.replace('_', ' ')
+    dateString = dateString.replace('_', ' ').strip()
```

Comment:
This is fine, but I think the `.strip()` was previously in the calling code and you deleted it by mistake.  Definitely doesn't hurt if this function handles dates with extra whitespace.